plan/plans/info.go

```diff
@@ -361,8 +361,14 @@ func (isp *InfoSchemaPlan) fetchColumns(schemas []*model.DBInfo) {
 				if decimal == types.UnspecifiedLength {
 					decimal = 0
 				}
-				dataType := types.TypeToStr(col.Tp, col.Charset == charset.CharsetBin)
-				columnType := fmt.Sprintf("%s(%d)", dataType, colLen)
+				columnType := types.TypeToStr(col.Tp, col.Charset == charset.CharsetBin)
+				if col.Decimal == types.UnspecifiedLength {
```

Comment:
Yes.
And for data type without decimal, n will always be UnSpecifiedLength until the data is sent to mysql-client.